### PR TITLE
`Exam mode`: Fix live monitoring issues on Chrome and sorting of import dialog

### LINKED ISSUE #5716: `Exam live statistics`: Chrome issue (Error: <path> attribute d: Expected number)
### Describe the bug

The Exam Live Statistics system causes errors when using the Chrome browser (these errors do not occur with Firefox).

image (4)

Possible reasons:
- Decimals with multiple decimal places (precision)
- Updating data during re-rendering
- Malformed data: https://stackoverflow.com/questions/62245818/error-path-attribute-d-expected-number-521739130434781-nanl60-739130434
- ...

### To Reproduce

1. Go to Exam Live Statistics Page with the Chrome browser
2. Participate with multiple students
3. Open the console
4. See errors

### Expected behavior

Similar to other browsers, the exam live statistics system should not produce any issues or errors in Chrome.

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by stefanwaldhauser**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Fixes issue no. 17, and no.22 (Closes #5716) found during [exam mode testing](https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=138610767).

### Description

Bug No 17: Clicking on the sort arrows in the import modal for an exam results in an error.
Problem: 
- The exam mode ordering was broken because the sorted column was `exam.testExam` instead of just `testExam`.
- The course title ordering was broken as DISTINCT was used. 

Bug No 22: Exam Monitoring page broken on chrome because of thousand of errors.
Problem: Same Error as described previously here;
https://github.com/ls1intum/Artemis/pull/4479
```
You might notice that the appearance of these bar charts differ a bit from other implementations with ngx-charts. The edges of the bars are not rounded here. 

This is due to an error output I was facing while testing this implementation locally (to be more precise, this error only occurs when the "show (un-)rated results" button is triggered). 

d3 seems to have issues when it comes to changing bars representing a non-zero value to zero. 

It is a bit weird because this error occurred for all charts except the point distribution chart (where the value 0 occurs as well). I did some research and found 

https://github.com/swimlane/ngx-charts/issues/498. There, disabling the edge rounding was proposed as workaround, which worked for me.
```
Solution: Deactivated rounding of bar charts in exam monitoring and all other instances of bar charts I found in the codebase

### Steps for Testing
Bug 17: Try to import an exam and sort by exam mode property. This should work and sorting by course title should be deactivated
Bug 22: Test that exam monitoring page is not broken on Chrome (open console)

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
- No UI changes

### Screenshots
- No UI changes

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/manage/grading/charts/sca-category-distribution-chart.component.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ enum ScaChartBarTitle {
                     [yAxis]="true"
                     [xAxisTickFormatting]="xAxisFormatting"
                     (select)="onSelect($event)"
-                >
+=                >
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/programming/manage/grading/charts/sca-category-distribution-chart.component.ts:41]: This change seems not intentional.

Comment (stefanwaldhauser) [reply src/main/webapp/app/exercises/programming/manage/grading/charts/sca-category-distribution-chart.component.ts]: Good catch! Removed the mistake in e1fbbe43e28ea056f1c428c41a72e8deed2f16d6

**Review by Strohgelaender — DISMISSED** [review]
code changes look reasonable, but I did not test this PR.

File: src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java [hunk 2]
```diff
@@ -461,7 +461,7 @@ public void setExamArchivePath(String examArchivePath) {
      */
     public enum ExamSearchColumn {
 
-        ID("id"), TITLE("title"), COURSE_TITLE("course.title"), EXAM_MODE("exam.testExam");
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java:464]: Why do we not support sorting based on the course title?

Comment (ge65cer) [reply src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java:464]: Hmm… it works for exercises and lectures but not for exams?

**Review by ge65cer — DISMISSED** [review]
Chrome bug is fixed.

**Review by simon-hng — DISMISSED** [review]
code good

File: src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java (hunk 2 above)
Comment (stefanwaldhauser) [reply src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java]: @ge65cer I now fixed it by removing the "DISTINCT" instead in 810237674bb7a0cf3e3100a075e78e2e3ff410c0. Now sorting by course title also works :)

**Review by ge65cer — APPROVED** [review]
Nice, I was sure something else must be causing the issue.

**Review by JohannesStoehr — APPROVED** [review]
Changes look good to me.
One small thing, you sometimes added ` [roundEdges]="false"` to the bottom and sometimes to the top of the parameters of the chart-component

**Pull request merged**